web-registry/src/features/ecocredit/CreateBatchBySteps/CreateBatchMultiStepForm/CreditBasics.tsx

```diff
@@ -0,0 +1,265 @@
+import React, { useState } from 'react';
+import { useFormikContext, Field, FieldArray } from 'formik';
+import * as Yup from 'yup';
+import { Box, IconButton, Link } from '@mui/material';
+import { useTheme, DefaultTheme as Theme } from '@mui/styles';
+
+import { Body } from 'web-components/lib/components/typography';
+import InputLabel from 'web-components/lib/components/inputs/InputLabel';
+import OnBoardingCard from 'web-components/lib/components/cards/OnBoardingCard';
+import TextField from 'web-components/lib/components/inputs/TextField';
+import { DatePickField } from 'web-components/lib/components/inputs/DatePickField';
+import OutlinedButton from 'web-components/lib/components/buttons/OutlinedButton';
+import TrashIcon from 'web-components/lib/components/icons/TrashIcon';
+import { AddCertificationModal } from 'web-components/lib/components/modal/AddCertificationModal';
+import {
+  requiredMessage,
+  invalidDate,
+  vcsRetirementSerialRE,
+  invalidVCSRetirement,
+  invalidVCSID,
+  numericOnlyRE,
+  isValidJSON,
+  invalidJSON,
+} from 'web-components/lib/components/inputs/validation';
+import { NameUrl } from 'web-components/lib/types/rdf';
+import { VCSBatchMetadataLD } from 'web-components/lib/types/rdf/C01-verified-carbon-standard-batch';
+import { Option } from 'web-components/lib/components/inputs/SelectTextField';
+
+import {
+  ProjectSelect,
+  CreditClassSelect,
+  MetadataJSONField,
+} from '../../../../components/molecules';
+
+export interface CreditBasicsFormValues {
+  classId: string;
+  startDate: Date | null;
+  endDate: Date | null;
+  metadata?: Partial<VCSBatchMetadataLD> | string | undefined;
+}
+
+const vcsMetadataSchema: Yup.AnyObjectSchema = Yup.object({
+  'regen:vcsProjectId': Yup.string()
+    .required(requiredMessage)
+    .matches(numericOnlyRE, { message: invalidVCSID }),
+  'regen:vcsRetirementSerialNumber': Yup.string()
+    .required(requiredMessage)
+    .matches(vcsRetirementSerialRE, { message: invalidVCSRetirement }),
+});
+
+const JSONSchema = Yup.string().test(
+  'is-json',
+  invalidJSON,
+  (value, context) => !!value && isValidJSON(value),
+);
+
+export const validationSchemaFields = {
+  classId: Yup.string().required(requiredMessage),
+  startDate: Yup.date().required(requiredMessage).typeError(invalidDate),
+  endDate: Yup.date().required(requiredMessage).typeError(invalidDate),
```

Comment:
I believe we could also limit this end date to be in the past cc/ @S4mmyb

Reply:
@S4mmyb is this confirmed? should the end date be before the present day?

Reply:
and probably also make sure the startDate before endDate if that's possible to do so with yup